src/test/java/com/github/tomakehurst/wiremock/matching/StringValuePatternTest.java

```diff
@@ -15,40 +15,80 @@
  */
 package com.github.tomakehurst.wiremock.matching;
 
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.reflect.ClassPath;
-import java.lang.reflect.Constructor;
+import java.io.*;
 import java.lang.reflect.Modifier;
-import java.util.Arrays;
-import java.util.Set;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
 import org.junit.jupiter.api.Test;
 
-public class StringValuePatternTest {
+@SuppressWarnings({"unchecked", "rawtypes"})
+class StringValuePatternTest {
 
   @Test
-  public void allSubclassesHaveWorkingToString() throws Exception {
-    Set<ClassPath.ClassInfo> allClasses =
-        ClassPath.from(Thread.currentThread().getContextClassLoader()).getAllClasses();
+  void allSubclassesHaveWorkingToString() throws Exception {
+    Set<Class> matchingClasses = getClassOfPackage("com.github.tomakehurst.wiremock.matching");
 
-    allClasses.stream()
-        .filter(
-            classInfo ->
-                classInfo.getPackageName().startsWith("com.github.tomakehurst.wiremock.matching"))
-        .map(
-            input -> {
-              try {
-                return input.load();
-              } catch (Throwable e) {
-                return Object.class;
-              }
-            })
-        .filter(clazz -> clazz.isAssignableFrom(StringValuePattern.class))
-        .filter(clazz -> !Modifier.isAbstract(clazz.getModifiers()))
-        .forEach(this::findConstructorWithStringParamInFirstPosition);
+    var matchingClassesAfterFilter01 =
+        matchingClasses.stream()
+            .filter(clazz -> clazz.isAssignableFrom(StringValuePattern.class))
+            .collect(Collectors.toSet());
+
+    var matchingClassesAfterFilter02 =
```

Comment:
Any reason we can't collapse all this down to a single stream rather than going via 2 Sets since we finish with a forEach?